website_user_login_redirect_custom/controllers/main.py

```
@@ -0,0 +1,77 @@
+        # Do not redirect backend/internal users
+        if request.env.user.has_group("base.group_user"):
+            return False
+
+        enabled = (
+            request.env["ir.config_parameter"]
+            .sudo()
+            .get_param("website_user_login_redirect_custom.enabled", "0")
+        )
+        return enabled == "1"
+
+    def _get_custom_redirect_url(self):
```

Ruff (0.14.0)

Missing return type annotation for private function `_get_custom_redirect_url`

(ANN202)